scripts/release/goa_prepare/goa_submission.pl

```diff
@@ -274,59 +277,118 @@ sub get_reaction_id {
+
+    if ($ontology_letter eq 'F') {
+        push @reasons_to_exclude, "No active unit and physical entity is a complex, set or, polymer: $catalyst_id"
+            if (scalar @active_units == 0) && ($physical_entity->is_a('Complex') || $physical_entity->is_a('EntitySet') || $physical_entity->is_a('Polymer'));
+        push @reasons_to_exclude, "Active unit is a complex or polymer: $catalyst_id"
+            if (scalar @active_units == 1) && ($active_units[0]->is_a('Complex') || $active_units[0]->is_a('Polymer'));
+        push @reasons_to_exclude, "Active unit is a set with non-EWAS members: $catalyst_id"
+            if (scalar @active_units == 1) && $active_units[0]->is_a("EntitySet") && !set_has_only_EWAS_members($active_units[0]);
+        push @reasons_to_exclude, "Multiple active units: $catalyst_id"
+            if scalar @active_units > 1;
+    }
+    
+    return @reasons_to_exclude;
+}
+
+sub get_proteins_for_molecular_function {
+    my $physical_entity = shift;
+    return unless $physical_entity;
     
-    push @errors, "No physical entity: $catalyst_id" if (!$physical_entity);
-    push @errors, "No compartment for physical entity: $catalyst_id" unless $physical_entity && $physical_entity->compartment->[0];
-    push @errors, "No active unit and physical entity is a complex with multiple, distinct proteins: $catalyst_id"
-        if (scalar @active_units == 0) && (scalar @{find_rps($catalyst_activity)} > 1) && $physical_entity->is_a('Complex');
-    push @errors, "Active unit is a complex: $catalyst_id" if (scalar @active_units == 1) && $active_units[0]->is_a('Complex');
-    push @errors, "Multiple active units: $catalyst_id" if scalar @active_units > 1;
+    my $logger = get_logger(__PACKAGE__);
+    if ($physical_entity->is_a('Complex') || $physical_entity->is_a('Polymer')) {
+        $logger->warn($physical_entity->displayName . ' (' . $physical_entity->db_id . ') is a complex or polymer and should not be used to obtain protein annotations');
+        return;
+    }
     
-    return @errors;
+    my @proteins = ();
+    if ($physical_entity->is_a('EntitySet') && set_has_only_EWAS_members($physical_entity)) {
+        push @proteins, @{$physical_entity->hasMember};
+    } elsif ($physical_entity->is_a('EntityWithAccessionedSequence')) { # If the entity is a protein
+        push @proteins, $physical_entity;
+    }
+    return @proteins;
+}
+
+sub get_proteins_for_biological_process {
+    my $physical_entity = shift;
+
+    return get_proteins_from_physical_entity($physical_entity);
 }
 
 sub get_proteins_from_physical_entity {
     my $physical_entity = shift;
 
     return unless $physical_entity;
-        
+
     my @proteins = ();
     if ($physical_entity->is_a('Complex') || $physical_entity->is_a('EntitySet') || $physical_entity->is_a('Polymer')) {
         foreach my $sub_element (@{$physical_entity->hasMember}, @{$physical_entity->hasComponent}, @{$physical_entity->repeatedUnit}) {
```

Comment:
This won't cause issues if the attribute is an invalid attribute for an instance? For example, Complex/EntitySet and RepeatedUnit? This would cause problems if it was a Java equivalent, I'm pretty sure. Additionally, I notice that Complexes/EntitySets and their super, PhysicalEntity have an attribute called 'RepeatedUnitOf'. How is that attribute used in Reactome?

Reply:
Nope, for the Perl API if the attribute doesn't exist for the instance then the value gets de-referenced to an empty array.